app/views/settings/hostings/show.html.erb

```diff
@@ -80,6 +80,32 @@
           </div>
         </div>
       </div>
+
+      <div>
+        <h2 class="font-medium mb-1"><%= t(".invite_settings.title") %></h2>
+        <p class="text-gray-500 text-sm mb-4"><%= t(".invite_settings.description") %></p>
+        <div class="flex cursor-pointer items-center justify-between">
+          <div>
+            <p><%= t(".invite_settings.require_invite_for_signup") %></p>
+          </div>
+          <div class="relative inline-block select-none">
+            <%= form.check_box :require_invite_for_signup, class: "sr-only peer", "data-auto-submit-form-target": "auto", "data-autosubmit-trigger-event": "input" %>
+            <%= form.label :require_invite_for_signup, "&nbsp;".html_safe, class: "maybe-switch" %>
+          </div>
+          <div>
+            <% if Setting.require_invite_for_signup %>
+              <%= link_to invite_codes_path,
+                          class: "flex gap-1 bg-gray-50 text-gray-900 text-sm font-medium rounded-lg px-3 py-2",
+                          data: { turbo_frame: "invite_codes", turbo_stream: true }  do %>
+                <%= lucide_icon "binary", class: "w-5 h-5 text-gray-500" %>
+                <span><%= t(".invite_settings.generate_tokens") %></span>
+              <% end %>
+            <% end %>
+          </div>
+        </div>
+      </div>
+
+      <%= turbo_frame_tag "invite_codes" %>
```

Comment:
If we make a slight modification to this, we can avoid the custom `respond_to`.  This would fetch the index frame and render it here:

```
<%= turbo_frame_tag :invite_codes, src: invite_codes_path %>
```

When we do this, we'll need to update the button above to make a post request to `invite_codes_path` to invoke a CREATE action.

That way, we separate the responsibilities a bit better:

- Index - simply renders generated codes, no generation step
- Create - creates one code at a time, on-demand

Reply:
[68a069d](https://github.com/maybe-finance/maybe/pull/1163/commits/68a069d9d451844e1cefdbae152a9ffea6dafa4f)